### ISSUE TITLE
AveragePool fails when output width * height is not a multiple of 4

### ISSUE THREAD

**Issue by pixelspark** [haixuanTao/wonnx#32]
**Describe the bug**

When you feed AveragePool an NxCxWxH tensor where the output WxH (which depends on the kernel size) is not divisible by four, the following error occurs:

````
Shader error:
error: expected ')', found 'u'
   ┌─ wgsl:25:23
   │
25 │         if (gidx < 4.5u) {
   │                       ^ expected ')'
````

The relevant part of the template (templates/pool/aggregate.wgsl):

````wgsl
if (gidx < {{ o_lens[0] / 4 }}u) {
````

I ran into this when implementing `GlobalAveragePool` which is basically `AveragePool` with the kernel size equal to the image size, i.e. simply averaging to a single number per channel (NxCxWxH -> NxCx1x1).

**To Reproduce**

Use AveragePool in a way that sets `o_lens[0]` to something not divisible by 4.

**Expected behavior**

This should just work - the output may of course be a vector of length rounded up to the next multiple of 4, that is automatically chopped of if it's the output vector (or not relevant when this is an input to the next op).

**Comment by pixelspark** [comment]
This can be tested fairly easily using [this test](https://github.com/haixuanTao/wonnx/pull/34/files#diff-16611a4a68494d7708e6def00fcee2905e9387a44c541cd177bc72c30d4509ec), by changing `channels`  from `4` to `2` and removing the last two elements of the result vector. The test should then still pass but it doesn't.

**Comment by haixuanTao** [comment]
Yeah, I wanted to make a non integer division as the logic would not stand for non 4 divisible pooling. 

The thing is that using vec4 is much better performance wise than other things... But I agree that for resilience an alternative for non 4 divisible lengths should be implemented.
